contracts/JuiceBuybackDelegate.sol

```diff
@@ -318,15 +319,28 @@ contract JuiceBuybackDelegate is IJBFundingCycleDataSource, IJBPayDelegate, IUni
         _preferClaimedTokens: true
       });
 
-      // Mint the reserved token straight to the reserve
+      // Mint the reserved token with this address as beneficiary
       controller.mintTokensOf({
         _projectId: _data.projectId,
         _tokenCount: _amountReceived - _nonReservedToken,
         _beneficiary: _data.beneficiary,
         _memo: _data.memo,
-        _preferClaimedTokens: _data.preferClaimedTokens,
+        _preferClaimedTokens: false,
         _useReservedRate: true
         });
+
+      // Burn the non-reserve token which are in this address
+      // TOD0: use arithmetic magic instead of balanceOf
+      uint256 _nonReservedTokenInContract = projectToken.balanceOf(address(this));
+
+      if(_nonReservedTokenInContract != 0)
```

Comment:
just wondering if there would be a scenario with nonReserved Bal in contract > 0 since
`uint256 _reservedToken = _amountReceived - _nonReservedToken` and we transfer the complete `_nonReservedToken` to the `beneficiary`

Reply:
Hmm, weird, github display code which changed since then -> I removed the call to balanceOf before the review, as, indeed, the contract's balance should be 0 (no rounding error possible iic)

Reply:
Yeah I'm just wondering since it'll be 0 always we probably don't need the non reserved burn block

Reply:
This is my current version: 
image.png

Reply:
this is, for me, the current version:
![image](https://user-images.githubusercontent.com/83670532/208934190-6a45c1ea-9780-4573-937f-e78a8b71c30c.png)

Reply:
Yeah I just commented based on git blame 😅
What I meant was I was wondering if there gonna be any scenario where nonReservedTokenInContract is > 0 ?

Reply:
Yes, this would be most of the time the case, as the idea is to burn the amount to add to the reserve, mint with the reserved rate *to the contract* then burn what the contract actually received (ie having a net effect only on the reserve)